File: it/versions/structures/default/python3/bar/v1.py
class Other:
  def __init__(self, name):
    self.name = name

  def get_name(self):
    return self.name

  @staticmethod
  def decode(data):
    f_name = data["name"]

    if not isinstance(f_name, str):
      raise Exception("not a string")

    return Other(f_name)

  def encode(self):
    data = dict()

    if self.name is None:
      raise Exception("name: is a required field")

    data["name"] = self.name

    return data

  def __repr__(self):
    return "<Other name:{!r}>".format(self.name)

File: it/versions/structures/default/dart/bar/v1.dart
class Other {
  String name;

  Other(
    this.name
  );

  static Other decode(dynamic _dataDyn) {
    if (!(_dataDyn is Map<String, dynamic>)) {
      throw 'expected Map<String, dynamic>, but got: $_dataDyn';
    }

    Map<String, dynamic> _data = _dataDyn;

    var name_dyn = _data["name"];
    if (name_dyn == null) {
      throw "expected value but was null";
    }
    if (!(name_dyn is String)) {
      throw 'expected String, but was: $name_dyn';
    }
    final String name = name_dyn;

    return Other(name);
  }

  Map<String, dynamic> encode() {
    Map<String, dynamic> _data = Map();

    _data["name"] = this.name;

    return _data;
  }
}
